src/main/java/io/knowledgeassets/myskills/server/user/profile/UserProfileService.java

```diff
@@ -0,0 +1,240 @@
+
+	private static final SimpleDateFormat SDF = new SimpleDateFormat("dd.MM.yyyy");
+
+	private static final ClassPathResource DEFAULT_TEMPLATE_RESOURCE = new ClassPathResource("templates/user-profile.docx");
+
+	private final UserRepository userRepository;
+	private final UserSkillRepository userSkillRepository;
+	private final String templatePath;
+
+	public UserProfileService(UserRepository userRepository,
+							  UserSkillRepository userSkillRepository,
+							  @Value("${templates.user.profile.path:#{null}}") String templatePath) {
+		this.userRepository = requireNonNull(userRepository);
+		this.userSkillRepository = requireNonNull(userSkillRepository);
+		this.templatePath = templatePath;
+	}
+
+	/**
+	 * Retrieves MS Word document (DOCX) with anonymous user profile data.
+	 * @param referenceId user reference id
+	 * @return MS Word document with anonymous user profile data
+	 */
+	public InputStream getAnonymousUserProfileDocument(String referenceId) {
+
+		final User user = getUserByReferenceId(referenceId);
+
+		try (final InputStream is = getTemplate();
+			 final XWPFDocument document = new XWPFDocument(is)) {
+			replacePlaceholders(document, user);
+			final ByteArrayOutputStream b = new ByteArrayOutputStream();
+			document.write(b);
+			return new ByteArrayInputStream(b.toByteArray());
+		}
+		catch (IOException | EmptyFileException | NotOfficeXmlFileException e) {
+			throw new UserProfileDocumentException(format("An error has occurred when building user " +
+					"profile document for a user with the reference id %s", referenceId), e);
+		}
+	}
+
+	private InputStream getTemplate() throws IOException {
+		if (StringUtils.isEmpty(this.templatePath)) {
+			return DEFAULT_TEMPLATE_RESOURCE.getInputStream();
+		}
+		else {
+			final Path path = Paths.get(this.templatePath);
+			if (path.toFile().exists()) {
+				try {
+					return new FileSystemResource(path).getInputStream();
+				}
+				catch (IOException ex) {
+					log.warn(format("The template file \"%s\" cannot be read. The default template file will be used as a fallback option.", path), ex);
+					return DEFAULT_TEMPLATE_RESOURCE.getInputStream();
+				}
+			}
+			else {
+				log.warn(format("The template file \"%s\" does not exist. The default template file will be used as a fallback option.", path));
+				return DEFAULT_TEMPLATE_RESOURCE.getInputStream();
+			}
+		}
+	}
+
+	private User getUserByReferenceId(String referenceId) {
+		return userRepository.findByReferenceId(referenceId).orElseThrow(() -> {
+			final String[] searchParamsMap = {"referenceId", referenceId};
+			return NoSuchResourceException.builder()
+					.model(Model.USER)
+					.searchParamsMap(searchParamsMap)
+					.build();
+		});
+	}
+
+	private void replacePlaceholders(XWPFDocument document, User user) {
+		replaceTablesPlaceholders(document, user);
+		replaceParagraphsPlaceholders(document, user, document.getParagraphs());
+		ofNullable(document.getFooterList()).ifPresent(footers ->
```

Comment:
Again a matter of taste... personally, I hate such an expression because I find `if (document.getFooterList() != null)` a lot more readable at a glance. Maybe it's because I'm a little bit old-fashioned. 😉 You can leave it like this if it feels right for you. 😄 As an argument for my point of view: Even developers from the JDK team discourage instantiating `Optional` instances just to chain methods like `ifPresent` because this is not what `Optional` was made for. 😉